File: app/src/main/java/com/example/movies/ui/detailtabs/view/RecommendedFragment.java
package com.example.movies.ui.detailtabs.view;

import android.os.Bundle;

import com.example.movies.core.base.BasePageableFragment;
import com.example.movies.core.base.BaseSimilarPageableFragment;


public class RecommendedFragment extends BasePageableFragment {

    @Override
    public void fetchRequest() {
        service.fetchRecommendedMovie(id, this, pagination);
    }

    public static RecommendedFragment newInstance(int id) {
        RecommendedFragment fragment = new RecommendedFragment();
        Bundle bundle = new Bundle();
        bundle.putInt("id", id);
        fragment.setArguments(bundle);
        return fragment;
    }


}

File: app/src/main/java/com/example/movies/ui/detailtabs/view/RecommendedFragment.kt
package com.example.movies.ui.detailtabs.view

import android.os.Bundle

import com.example.movies.core.base.BasePageableFragment
import com.example.movies.core.base.BaseSimilarPageableFragment


class RecommendedFragment : BasePageableFragment() {

    override fun fetchRequest() {
        movieId?.let {id ->
            service.fetchRecommendedMovie(id, this, pagination)
        }
    }

    companion object {

        fun newInstance(id: Int): RecommendedFragment {
            val fragment = RecommendedFragment()
            val bundle = Bundle()
            bundle.putInt("id", id)
            fragment.arguments = bundle
            return fragment
        }
    }


}
